snapcraft/internal/cache.py

```diff
@@ -0,0 +1,80 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import shutil
+
+from xdg import BaseDirectory
+
+import snapcraft
+
+
+logger = logging.getLogger(__name__)
+
+
+class SnapcraftCache:
+    """Generic cache base class.
+
+    This class is responsible for cache location, notification and pruning.
+    """
+    def __init__(self):
+        self.cache_root = os.path.join(
+            BaseDirectory.xdg_cache_home, 'snapcraft')
+        self.project_cache_root = os.path.join(
+            self.cache_root, snapcraft.internal.load_config().data['name'])
```

Comment:
Can this be moved to a `SnapcraftProjectCache` inheriting from this class? This would make moving the apt caching logic cleaner (as it is a global cache).